### ISSUE TITLE
[Feature Request] Resizable child windows

### ISSUE THREAD

**Issue by Arcnor** [cycfi/elements#419]
Currently child windows cannot be resized, only moved and/or closed. My assumption is that a `resizeable` "trait" (sorry if that's not the right term!) can be added, similar to the `moveable` one.

**Comment by djowel** [comment]
> My assumption is that a `resizeable` "trait" (sorry if that's not the right term!) can be added, similar to the `moveable` one.

Yes, that is correct.

**Comment by djowel** [comment]
I'm working in this.

**Comment by djowel** [comment]
Ah, then minimize/maximize have to be added.

**Comment by djowel** [comment]
Implemented now in `master` and `skia_2024` branches.

**Comment by Arcnor** [comment]
I've finally tested this, seems to work mostly ok, although (and talking about the "Child Windows" example) depending on how far into the window the mouse cursor is, it either resizes the window or triggers the scrollbar behavior (when talking about the bottom/right areas) / moves the window (when talking about the top area), but the mouse cursor is still the resize one.

There is probably a difference between the area that triggers the cursor change and the area that actually effects the resizing behavior.

**Comment by djowel** [comment]
> I've finally tested this, seems to work mostly ok, although (and talking about the "Child Windows" example) depending on how far into the window the mouse cursor is, it either resizes the window or triggers the scrollbar behavior (when talking about the bottom/right areas) / moves the window (when talking about the top area), but the mouse cursor is still the resize one.
> 
> There is probably a difference between the area that triggers the cursor change and the area that actually effects the resizing behavior.

Right! Good observation. The active areas overlap. What I should do is move the resize area outside the scroll bar's active area. I'll see what I can do!

**Comment by djowel** [comment]
Try it now. `master` and `skia_2024`

**Comment by djowel** [comment]
Need more cursors. I added two cursors for the corner sizing (north-to-east and south-to-west) for the Mac, example:

<img width="226" alt="Screenshot 2024-07-14 at 1 24 07 PM" src="https://github.com/user-attachments/assets/062f8f24-08b4-4d79-89a9-609048c24757">

I also fixed the correct "sizing" cursors on the Mac.

If you can add the switches for setting these new cursors for GTK and Windows, that would be great. If not, I'll have this open and get back to it later. See:

```
void set_cursor(cursor_type type)
```

in base_view.cpp

**Comment by Arcnor** [comment]
Awesome. I actually noticed that corners were missing as well, but didn't want to bother you anymore 🙂.

I'll check and test Windows corner cursors (it's `IDC_SIZENWSE` and `IDC_SIZENESW` there), will also check GTK docs but not sure I'll be able to test the implementation, will let you know.

**Comment by Arcnor** [comment]
So I've implemented and tested Windows, the corner cursor seems to need a bit of tinkering. The area in which the cursor appear seems to be very small, and also the cursor can go from "h cursor" -> "diagonal cursor" -> "h cursor" (for example) when slowly panning the cursor from the right side of the window until you get to the corner. Same with the "v cursor".

As for GTK3 I'm having a bit of a hard time finding the diagonal cursor. I see there should be one on this image (although this might be GTK2?): ![](https://i.imgur.com/J1IdzKD.png)

However, I can't find any diagonal cursors on the docs: https://docs.gtk.org/gdk3/enum.CursorType.html

**Comment by djowel** [comment]
> So I've implemented and tested Windows, the corner cursor seems to need a bit of tinkering. The area in which the cursor appear seems to be very small, and also the cursor can go from "h cursor" -> "diagonal cursor" -> "h cursor" (for example) when slowly panning the cursor from the right side of the window until you get to the corner. Same with the "v cursor".

For sure. Right now the margin is just 5px. You are welcome to make some adjustments.

**Comment by djowel** [comment]
Did some refactoring and renames.

**Comment by djowel** [comment]
I'm going to move this to a feature branch (resizable_window and resizable_window_skia) until it is complete.

**Comment by djowel** [comment]
Done. This is moved to the `resizable_window` and `resizable_window_skia` branches now.
